workflow/controller/controller.go

```diff
@@ -194,6 +205,45 @@ func (wfc *WorkflowController) Run(ctx context.Context, wfWorkers, podWorkers in
 	<-ctx.Done()
 }
 
+// Create and initialize the Synchronization Manager
+func (wfc *WorkflowController) createSynchronizationManager() error {
+	syncLimitConfig := func(lockName string) (int, error) {
+		items := strings.Split(lockName, "/")
```

Comment:
Should this be a function of its own? Something like `func decodeLockName(lockName string) lockName` where

```go
type lockName struct {
  field0 string
  field1 string
  ...
}
```

Using strings splits to serialize and de serialize data could bring bugs